releases/stable-branches-and-release-cycle.md

```diff
@@ -0,0 +1,79 @@
+# Producing and supporting stable releases
+
+## Summary
+
+Today the Submariner project releases one minor version a month without capacity
+for stable branches or bugfix releases. In the following proposal we define a plan
+to make this possible starting with 0.8.
+
+## Proposal
+
+### Alignment of branch -> image releases
+
+We produce currently `<image>:devel` for every master push, along with `<image>:x.y.z` for
+the releases.
+
+Changing the `master` branch name to `devel` would make this homogeneous so our image releases
+are simply `<image>:<branch-or-tag>`.
+
+### Creating one stable branch for every minor release
+
+Approaching the release of every minor or major version a new stable branch will be forked
+off `devel` branch in the form of `release-x.y` for every existing repository.
+
+Any changes related to pinning shipyard base image will be done in that branch.
+
+At that point in time, any patches which belong to the release will need to be merged in master
```

Comment:
```suggestion
Once the branch exists, any patches which should be part of the release will need to be merged in `devel`
```

Reply:
I think patches should be created in `devel` first and then cherry pick anywhere else needed

Reply:
Yet, but this mean, what to do with patches that are bugfixes to make the release possible.

Reply:
They still need to go through the main branch. We ran into lots of problems before we enforced this rule in ODL: in quite a few instances, fixes were made on release branches, but forgotten in the main branch, so the next release failed again for all the same reasons...